### PR TITLE
Add musl cross-compilers for x86 and aarch64 linux

### PR DESCRIPTION
**Pull request by nicksieger**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -73,19 +75,26 @@ RUN curl -sSL -H'Authorization: Bearer QQ==' "https://ghcr.io/v2/homebrew/core/l
 		"libtool/${LIBTOOL_VERSION}/include/" \
 		"libtool/${LIBTOOL_VERSION}/lib/"
 
-FROM osx-cross-base AS final
+FROM cross-base AS musl-cross
+WORKDIR "${MUSL_CROSS_PATH}"
+RUN git clone https://github.com/richfelker/musl-cross-make.git .
+RUN make install TARGET=aarch64-linux-musl OUTPUT=${MUSL_CROSS_PATH}/target
+RUN make install TARGET=x86_64-linux-musl OUTPUT=${MUSL_CROSS_PATH}/target
```
Comment (nicksieger) [inline Dockerfile:82]: Only installing x86 and arm64/aarch64 cross compilers for now

**Review by nicks — APPROVED** [review]
LGTM

this seems fine, though i assumed we were going to statically compile glibc. were you thinking we'd dynamically compile against musl?

**Comment by nicksieger** [comment]
After looking around a bit, statically compiling against musl sounded like the easier way to go. And now we can compile dynamic-linking executables for alpine too if we like.

**Review by milas — APPROVED** [review]

**Comment by nicksieger** [comment]
Trying one more round of changes. I think the way I installed both of the musl cross libraries in the same directory doesn't work, because I could only get a successful x86 build. The aarch64 build failed with a `collect2: fatal error: cannot find ld` error.

**Comment by nicksieger** [comment]
Switched out musl-cross-make built toolchains for pre-built ones at musl.cc. The ones I built were missing the ld.gold linker for some reason.

**Pull request merged**